soroban-sdk/src/crypto.rs

```diff
@@ -22,6 +22,28 @@ impl Crypto {
         unsafe { BytesN::unchecked_new(env.clone(), bin) }
     }
 
+    // Reseeds the pseudorandom number generator (PRNG) with the provided `seed` value.
+    pub fn prng_reseed(&self, seed: &Bytes) {
+        let env = self.env();
+        internal::Env::prng_reseed(env, seed.into()).unwrap_infallible();
+    }
+
+    // Returns a random u64 in the range between `lower` and `upper` inclusive.
+    pub fn u64_in_inclusive_range(&self, lower: u64, upper: u64) -> u64 {
+        let env = self.env();
+        internal::Env::prng_u64_in_inclusive_range(env, lower.into(), upper.into())
+            .unwrap_infallible()
+            .into()
+    }
+
+    // Shuffles a given vector v using the Fisher-Yates algorithm.
+    pub fn vec_shuffle(&self, v: VecObject) -> VecObject {
```

Comment:
I would pass in `v: Vec` instead, and call `env.check_same_env(v.env())` to make sure they from the same env.

Reply:
(See updated comment above on `check_same_env`)

Reply:
Ah that makes sense. Done 👍 -> [Pass a Vec to vec_shuffle instead of a VecObject](https://github.com/stellar/rs-soroban-sdk/pull/1023/commits/749973904a74135856edc45110f708df58eec2d0)